app/kubemci/pkg/gcp/healthcheck/healthchecksyncer.go

```diff
@@ -66,12 +69,30 @@ var _ HealthCheckSyncerInterface = &HealthCheckSyncer{}
 // EnsureHealthCheck ensures that the required health check exists.
 // Does nothing if it exists already, else creates a new one.
 // Returns a map of the ensured health checks keyed by the corresponding port.
-func (h *HealthCheckSyncer) EnsureHealthCheck(lbName string, ports []ingressbe.ServicePort, forceUpdate bool) (HealthChecksMap, error) {
+func (h *HealthCheckSyncer) EnsureHealthCheck(lbName string, ports []ingressbe.ServicePort, client kubernetes.Interface, forceUpdate bool) (HealthChecksMap, error) {
 	fmt.Println("Ensuring health checks")
 	var err error
 	ensuredHealthChecks := HealthChecksMap{}
 	for _, p := range ports {
-		hc, hcErr := h.ensureHealthCheck(lbName, p, forceUpdate)
+		ctx := ingresscx.NewControllerContext(client, p.SvcName.Namespace, 1*time.Second, false)
+		ki := kubeutils.NewKubernetesInformer(ctx)
```

Comment:
We dont need the full informer. Informer watches all services and pods and is a good model for controller that syncs multiple ingresses (all ingresses at some point). We use the informer to optimize such that we fetch resources only once and keep them in memory until we get a watch notification for change.

Here we are concerned with a single ingress, so we can fetch the service and pods directly for that ingress only. We know the services here. We can fetch pods corresponding to the services and check for their readiness probe. Since the health check is same for all, all pods are required to have the same readiness probe.

Reply:
👍 Yeah I was mostly concerned with reusing as much code as possible from ingress-gce, but you're right it's better to keep it simple here and just fetch directly.